### ISSUE TITLE
Form File field give me an error

### ISSUE THREAD

**Issue by enjoyjeremy** [getgrav/grav-plugin-form#25]
I have followed the instructions by copying the plugin config file to my user config folder and making sure it is enabled.
My field is set up as follows...
```yaml
- name: images
  label: Add one or more images
  type: file
   validate:
    required: true
```
But it's giving me this error when I try to upload a file (tested locally and on a server)
**Undefined index: files**

Here's the full error from the log file (with my local path omitted:

`
grav.CRITICAL: Undefined index: files - Trace: #0 [my local root]/user/plugins/form/classes/form.php(206): Whoops\Run->handleError(8, 'Undefined index...', '[my local root]/ww...', 206, Array) #1 [my local root]/user/plugins/form/classes/form.php(173): Grav\Plugin\Form->cleanFilesData('images', Array) #2 [my local root]/user/plugins/form/form.php(59): Grav\Plugin\Form->post() #3 [internal function]: Grav\Plugin\FormPlugin->onPageInitialized(Object(RocketTheme\Toolbox\Event\Event), 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #4 [my local root]/vendor/symfony/event-dispatcher/EventDispatcher.php(158): call_user_func(Array, Object(RocketTheme\Toolbox\Event\Event), 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #5 [my local root]/vendor/symfony/event-dispatcher/EventDispatcher.php(46): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #6 [my local root]/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php(23): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #7 [my local root]/system/src/Grav/Common/Grav.php(412): RocketTheme\Toolbox\Event\EventDispatcher->dispatch('onPageInitializ...', NULL) #8 [my local root]/system/src/Grav/Common/Grav.php(252): Grav\Common\Grav->fireEvent('onPageInitializ...') #9 [my local root]/index.php(37): Grav\Common\Grav->process() #10 {main} [] []
`

Is there anything else that needs to be set up or is there any issue with the plugin?

**Comment by w00fz** [comment]
Do you get this error when you try to submit without adding any file?

**Comment by enjoyjeremy** [comment]
Just tested and I actually get the error whether I add a file or not.

**Comment by w00fz** [comment]
Yeah that appeared to be an issue. The default settings were never pulled in. Could you try with the changes from 68e6c88 ?

**Comment by enjoyjeremy** [comment]
That fixed something :) but now I'm getting this error: 

**An exception has been thrown during the rendering of a template ("nl2br() expects parameter 1 to be string, array given") in "forms/default/data.txt.twig" at line 2.**

`
grav.CRITICAL: An exception has been thrown during the rendering of a template ("nl2br() expects parameter 1 to be string, array given") in "forms/default/data.txt.twig" at line 2. - Trace: #0 [mylocalroot]/vendor/twig/twig/lib/Twig/Template.php(355): Twig_Template->displayWithErrorHandling(Array, Array) #1 [mylocalroot]/cache/twig/08/08472d87ac792bb5cb798c52ae5daa7c74be7b278a12c2f6395e6a523434609c.php(23): Twig_Template->display(Array, Array) #2 [mylocalroot]/vendor/twig/twig/lib/Twig/Template.php(381): __TwigTemplate_41bf48ebaed551a0c226a1492280a21e9ba5fe489636e4abf2dbee33f005d0fb->doDisplay(Array, Array) #3 [mylocalroot]/vendor/twig/twig/lib/Twig/Template.php(355): Twig_Template->displayWithErrorHandling(Array, Array) #4 [mylocalroot]/cache/twig/58/58c2d6a11c74a9b41c547d78eb07e0bcd6e0eb64d10a1133088c9b3d5ed9da25.php(19): Twig_Template->display(Array) #5 [mylocalroot]/vendor/twig/twig/lib/Twig/Template.php(381): __TwigTemplate_be4f0749cdb71d52a9a3ae97da3842ee63461057762683897d01fcf0fba8c558->doDisplay(Array, Array) #6 [mylocalroot]/vendor/twig/twig/lib/Twig/Template.php(355): Twig_Template->displayWithErrorHandling(Array, Array) #7 [mylocalroot]/vendor/twig/twig/lib/Twig/Template.php(366): Twig_Template->display(Array) #8 [mylocalroot]/vendor/twig/twig/lib/Twig/Environment.php(347): Twig_Template->render(Array) #9 [mylocalroot]/system/src/Grav/Common/Twig/Twig.php(278): Twig_Environment->render('@Var:{% include...', Array) #10 [mylocalroot]/user/plugins/form/form.php(172): Grav\Common\Twig\Twig->processString('{% include 'for...', Array) #11 [internal function]: Grav\Plugin\FormPlugin->onFormProcessed(Object(RocketTheme\Toolbox\Event\Event), 'onFormProcessed', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #12 [mylocalroot]/vendor/symfony/event-dispatcher/EventDispatcher.php(158): call_user_func(Array, Object(RocketTheme\Toolbox\Event\Event), 'onFormProcessed', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #13 [mylocalroot]/vendor/symfony/event-dispatcher/EventDispatcher.php(46): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'onFormProcessed', Object(RocketTheme\Toolbox\Event\Event)) #14 [mylocalroot]/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php(23): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('onFormProcessed', Object(RocketTheme\Toolbox\Event\Event)) #15 [mylocalroot]/system/src/Grav/Common/Grav.php(412): RocketTheme\Toolbox\Event\EventDispatcher->dispatch('onFormProcessed', Object(RocketTheme\Toolbox\Event\Event)) #16 [mylocalroot]/user/plugins/form/classes/form.php(195): Grav\Common\Grav->fireEvent('onFormProcessed', Object(RocketTheme\Toolbox\Event\Event)) #17 [mylocalroot]/user/plugins/form/form.php(59): Grav\Plugin\Form->post() #18 [internal function]: Grav\Plugin\FormPlugin->onPageInitialized(Object(RocketTheme\Toolbox\Event\Event), 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #19 [mylocalroot]/vendor/symfony/event-dispatcher/EventDispatcher.php(158): call_user_func(Array, Object(RocketTheme\Toolbox\Event\Event), 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #20 [mylocalroot]/vendor/symfony/event-dispatcher/EventDispatcher.php(46): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #21 [mylocalroot]/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php(23): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #22 [mylocalroot]/system/src/Grav/Common/Grav.php(412): RocketTheme\Toolbox\Event\EventDispatcher->dispatch('onPageInitializ...', NULL) #23 [mylocalroot]/system/src/Grav/Common/Grav.php(252): Grav\Common\Grav->fireEvent('onPageInitializ...') #24 [mylocalroot]/index.php(37): Grav\Common\Grav->process() #25 {main} [] []
`

**Comment by enjoyjeremy** [comment]
That error only shows up when I add a file.

**Comment by enjoyjeremy** [comment]
I also get this error when I try to make changes to the settings through the Admin:
**Validation failed: files is not defined in blueprints**

**Comment by w00fz** [comment]
That is actually different. The files field stores data in an array format and the `data.txt.twig` doesn't support that. I know added the support for arrays in that twig but I really won't be of much use.
I suggest you save in a better yaml format, such as:

```yaml

**Comment by w00fz** [comment]
Regarding the admin, that is also an issue :) you can fix following the changes in 4054bcc

**Comment by enjoyjeremy** [comment]
Yep, that did it! Thanks!

**Comment by w00fz** [comment]
No problem, thanks for reporting the issue, good catch.

**Comment by rhukster** [comment]
Fixed in v1.0.2
